build.gradle

```diff
@@ -39,14 +39,32 @@ subprojects {
         }
     }
     dependencies {
-        implementation "com.google.guava:guava:29.0-jre"
-        implementation "org.apache.logging.log4j:log4j-core:2.14.0"
+        implementation "com.google.guava:guava:31.0.1-jre"
+        implementation "org.apache.logging.log4j:log4j-core:2.14.1"
         implementation "org.slf4j:slf4j-api:1.7.30"
-        implementation "org.apache.logging.log4j:log4j-slf4j-impl:2.14.0"
+        implementation "org.apache.logging.log4j:log4j-slf4j-impl:2.14.1"
         testImplementation("junit:junit:4.13") {
             exclude group: 'org.hamcrest' // workaround for jarHell
         }
+        constraints {
+            implementation('org.apache.httpcomponents:httpclient') {
+                version {
+                    require '4.5.13'
```

Comment:
QUES: how does it differ from implementation 'org.apache.httpcomponents:httpclient:4.5.13'?

Reply:
find the answer [here](https://docs.gradle.org/current/userguide/dependency_constraints.html). No issue.